src/renderer/screens/FirmwareUpdate.js

```diff
@@ -118,7 +117,22 @@ const FirmwareUpdate = (props) => {
   };
 
   const upload = async (options) => {
-    setFlashSteps(focusDeviceDescriptor.flashSteps(options));
+    let steps = [
```

Comment:
I'm thrilled that this code is no longer per-device, but the current implementation is.....hard to read.

Instead of having nested ifs and concatenation, I'd rather see a simpler if {} else if {} else if {}  else {}  structure.